libs/s25main/random/Random.h

```diff
@@ -139,3 +140,45 @@ class RandomFunctor
 
 /// Shortcut for creating an instance of RandomFunctor
 #define RANDOM_SHUFFLE(container) RandomFunctor::shuffleContainer(container, __FILE__, __LINE__)
+
+/// Return a random number in the interval [lowerBound, upperBound].
+template<typename RandomT>
+static unsigned getRandomNumber(const RandomT& rng, unsigned lowerBound, unsigned upperBound)
+{
+    std::uniform_int_distribution<typename RandomT::result_type> dist(lowerBound, upperBound);
```

Comment:
The result of the PRNG doesn't matter for the result if the distribution:
```suggestion
    std::uniform_int_distribution<unsigned> dist(lowerBound, upperBound);
```

Reply:
Why?

Reply:
You want an unsigned number that is uniformly distributed, don't you? The change makes it say that. Internally it could call the rng multiple times to get enough entropy to fill an unsigned. E.g. the rng could return only bytes.